src/System.Runtime/tests/System/IntPtr.cs

```diff
@@ -3,131 +3,163 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
-
+using System.Collections.Generic;
 using Xunit;
 
-public static unsafe class IntPtrTests
+public static class IntPtrTests
 {
     [Fact]
-    public static unsafe void TestBasics()
+    public static void TestZero()
     {
-        if (sizeof(void*) == 4)
-        {
-            // Skip IntPtr tests on 32-bit platforms
-            return;
-        }
+        VerifyPointer(IntPtr.Zero, 0);
+    }
 
-        IntPtr p;
-        int i;
-        long l;
+    [Fact]
+    public static void TestCtor_Int()
+    {
+        int i = 42;
+        VerifyPointer(new IntPtr(i), i);
+        VerifyPointer((IntPtr)i, i);
 
-        int size = IntPtr.Size;
-        Assert.Equal(size, sizeof(void*));
+        i = -1;
+        VerifyPointer(new IntPtr(i), i);
+        VerifyPointer((IntPtr)i, i);
+    }
+
+    [Fact]
+    public static void TestCtor_Long()
+    {
+        long l = 0x0fffffffffffffff;
+        VerifyPointer(new IntPtr(l), l);
```

Comment:
In my previous comments about 32-bit, I mentioned that some of the tests would still need to be specialized for 32-bit... this is an example... this will result in an OverflowException on 32-bit.  There are some others like it.